### PR TITLE
Add a New Random Encounter, the Road Tripper

### PR DESCRIPTION
**Pull request by MNG-cataclysm**
#### Summary
Content "Add a new random encounter, the road tripper."

#### Purpose of change
As part of a project requested by @I-am-Erk, I have created a new random encounter. This should also add some more variety to the world.

#### Describe the solution
Add a new random encounter that occurs at public washrooms. Players can find a traveling NPC in a camper van, going in a randomly selected cardinal direction for assorted reasons. This NPC will trade some items within the back of their camper in exchange for clean water, food, and gasoline at a premium, and can also point players to some points of interest if asked. Otherwise, most of the dialogue is small talk and minor lore. 

#### Describe alternatives you've considered
Not doing this.

#### Testing
Tested everything numerous times over the course of development, everything works like it should.

#### Additional context
This feature also suffers from a problem experienced by the Free Merchant caravans regarding item ownership, that's outlined in #71025. I don't know how to fix that, but any fixes made for that can be ported to this.

### CODE REVIEW

File: data/json/encounters/randec_independent_travelers.json [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        "weighted_list_eocs": [
+          [
+            {
+              "id": "going_north",
+              "effect": [
+                { "u_add_var": "going_north", "type": "dialogue", "context": "travel_direction", "value": "yes" },
+                { "u_add_var": "picked_direction", "type": "dialogue", "context": "travel_direction", "value": "yes" }
+              ]
+            },
+            { "const": 1 }
```
Comment (GuardianDll) [inline data/json/encounters/randec_independent_travelers.json:38]: weighted_list_eocs allows int
```suggestion
            1
```

File: data/json/encounters/randec_independent_travelers.json [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+          ],
+          [
+            {
+              "id": "going_south",
+              "effect": [
+                { "u_add_var": "going_south", "type": "dialogue", "context": "travel_direction", "value": "yes" },
+                { "u_add_var": "picked_direction", "type": "dialogue", "context": "travel_direction", "value": "yes" }
+              ]
+            },
+            { "const": 1 }
```
Comment (GuardianDll) [inline data/json/encounters/randec_independent_travelers.json:48]:
```suggestion
            1
```

File: data/json/encounters/randec_independent_travelers.json [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+          ],
+          [
+            {
+              "id": "going_east",
+              "effect": [
+                { "u_add_var": "going_east", "type": "dialogue", "context": "travel_direction", "value": "yes" },
+                { "u_add_var": "picked_direction", "type": "dialogue", "context": "travel_direction", "value": "yes" }
+              ]
+            },
+            { "const": 1 }
```
Comment (GuardianDll) [inline data/json/encounters/randec_independent_travelers.json:58]:
```suggestion
            1
```

File: data/json/encounters/randec_independent_travelers.json [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+          ],
+          [
+            {
+              "id": "going_west",
+              "effect": [
+                { "u_add_var": "going_west", "type": "dialogue", "context": "travel_direction", "value": "yes" },
+                { "u_add_var": "picked_direction", "type": "dialogue", "context": "travel_direction", "value": "yes" }
+              ]
+            },
+            { "const": 1 }
```
Comment (GuardianDll) [inline data/json/encounters/randec_independent_travelers.json:68]:
```suggestion
            1
```

File: data/json/npcs/random_encounters/camper_van_traveler.json [hunk 5]
```diff
@@ -0,0 +1,381 @@
...
+            { "npc_has_var": "knows_u", "type": "dialogue", "context": "first_meeting", "value": "yes" },
+            {
+              "not": { "npc_has_var": "gave_directions", "type": "dialogue", "context": "camper_van_traveler", "value": "yes" }
+            }
+          ]
+        },
+        "effect": [
+          {
+            "arithmetic": [ { "npc_val": "var", "var_name": "direction", "type": "randomize", "context": "dialogue" }, "=", { "rand": 4 } ]
+          },
```
Comment (GuardianDll) [inline data/json/npcs/random_encounters/camper_van_traveler.json:68]:
```suggestion
          { "math": [ "npc_randomize_dialogue_direction", "=", "rand(4)" ] },
```
I'm currently replacing this arithmetic comparisons with math, but it's WIP

File: data/json/npcs/random_encounters/camper_van_traveler.json [hunk 6]
```diff
@@ -0,0 +1,381 @@
...
+    "responses": [
+      { "text": "Let's take a look.", "topic": "TALK_NPC_CAMPER_VAN_TRAVELER", "effect": "start_trade" },
+      { "text": "<end_talking>", "topic": "TALK_DONE" }
+    ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_NPC_CAMPER_VAN_TRAVELER_INTERESTING_THINGS",
+    "dynamic_line": {
+      "compare_num": [ { "npc_val": "var", "var_name": "direction", "type": "randomize", "context": "dialogue" }, "=", { "const": 0 } ],
```
Comment (GuardianDll) [inline data/json/npcs/random_encounters/camper_van_traveler.json:176]:
```suggestion
      "math": [ "npc_randomize_dialogue_direction", "==", "0" ],
```

File: data/json/npcs/random_encounters/camper_van_traveler.json [hunk 7]
```diff
@@ -0,0 +1,381 @@
...
+      "yes": "I haven't found anything out of the ordinary, just a bunch of <zombies> and the ruins of our old lives.",
+      "no": {
+        "compare_num": [ { "npc_val": "var", "var_name": "direction", "type": "randomize", "context": "dialogue" }, "=", { "const": 1 } ],
```
Comment (GuardianDll) [inline data/json/npcs/random_encounters/camper_van_traveler.json:179]:
```suggestion
        "math": [ "npc_randomize_dialogue_direction", "==", "1" ],
```

File: data/json/npcs/random_encounters/camper_van_traveler.json [hunk 8]
```diff
@@ -0,0 +1,381 @@
...
+        "yes": "Actually, I came across a group of farmers on my way here.  They were a little suspicious, but turned out to be pretty nice.  Good place to buy some food, if they're still around.",
+        "no": {
+          "compare_num": [ { "npc_val": "var", "var_name": "direction", "type": "randomize", "context": "dialogue" }, "=", { "const": 2 } ],
```
Comment (GuardianDll) [inline data/json/npcs/random_encounters/camper_van_traveler.json:182]:
```suggestion
          "math": [ "npc_randomize_dialogue_direction", "==", "2" ],
```

File: data/json/npcs/random_encounters/camper_van_traveler.json [hunk 9]
```diff
@@ -0,0 +1,381 @@
...
+          "yes": "I found a small cabin out in the woods, someone was living there and had some serious chemistry equipment.  I managed to get good medical supplies from there.  They even had some oddball things too, like <swear> napalm and black powder.",
+          "no": {
+            "compare_num": [ { "npc_val": "var", "var_name": "direction", "type": "randomize", "context": "dialogue" }, "=", { "const": 3 } ],
```
Comment (GuardianDll) [inline data/json/npcs/random_encounters/camper_van_traveler.json:185]:
```suggestion
            "math": [ "npc_randomize_dialogue_direction", "==", "3" ],
```

File: data/json/npcs/random_encounters/camper_van_traveler.json [hunk 10]
```diff
@@ -0,0 +1,381 @@
...
+            "yes": "I ran in to someone living in an old barn, didn't want anything to do with the world.  The odd thing is, they looked like one of those old, Wild West cowboys.",
+            "no": {
+              "compare_num": [ { "npc_val": "var", "var_name": "direction", "type": "randomize", "context": "dialogue" }, "=", { "const": 4 } ],
```
Comment (GuardianDll) [inline data/json/npcs/random_encounters/camper_van_traveler.json:188]:
```suggestion
              "math": [ "npc_randomize_dialogue_direction", "==", "4" ],
```

File: data/json/npcs/random_encounters/camper_van_traveler.json [hunk 11]
```diff
@@ -0,0 +1,381 @@
...
+    },
+    "responses": [
+      {
+        "text": "How do I get to that farm?",
+        "topic": "TALK_FREE_MERCHANT_TEAMSTER_DIRECTIONS_2",
+        "condition": {
+          "and": [
+            {
+              "compare_num": [ { "npc_val": "var", "var_name": "direction", "type": "randomize", "context": "dialogue" }, "=", { "const": 1 } ]
+            },
```
Comment (GuardianDll) [inline data/json/npcs/random_encounters/camper_van_traveler.json:204]:
```suggestion
            { "math": [ "npc_randomize_dialogue_direction", "==", "1" ] },
```

File: data/json/npcs/random_encounters/camper_van_traveler.json [hunk 12]
```diff
@@ -0,0 +1,381 @@
...
+        "switch": true
+      },
+      {
+        "text": "How do I get to that cabin?",
+        "topic": "TALK_FREE_MERCHANT_TEAMSTER_DIRECTIONS_2",
+        "condition": {
+          "and": [
+            {
+              "compare_num": [ { "npc_val": "var", "var_name": "direction", "type": "randomize", "context": "dialogue" }, "=", { "const": 2 } ]
+            },
```
Comment (GuardianDll) [inline data/json/npcs/random_encounters/camper_van_traveler.json:223]:
```suggestion
            { "math": [ "npc_randomize_dialogue_direction", "==", "2" ] },
```

File: data/json/npcs/random_encounters/camper_van_traveler.json [hunk 13]
```diff
@@ -0,0 +1,381 @@
...
+        "switch": true
+      },
+      {
+        "text": "Where were they at?",
+        "topic": "TALK_FREE_MERCHANT_TEAMSTER_DIRECTIONS_2",
+        "condition": {
+          "and": [
+            {
+              "compare_num": [ { "npc_val": "var", "var_name": "direction", "type": "randomize", "context": "dialogue" }, "=", { "const": 3 } ]
+            },
```
Comment (GuardianDll) [inline data/json/npcs/random_encounters/camper_van_traveler.json:242]:
```suggestion
            { "math": [ "npc_randomize_dialogue_direction", "==", "3" ] },
```

File: data/json/npcs/random_encounters/camper_van_traveler.json [hunk 14]
```diff
@@ -0,0 +1,381 @@
...
+        "switch": true
+      },
+      {
+        "text": "Where was the bunker at?",
+        "topic": "TALK_FREE_MERCHANT_TEAMSTER_DIRECTIONS_2",
+        "condition": {
+          "and": [
+            {
+              "compare_num": [ { "npc_val": "var", "var_name": "direction", "type": "randomize", "context": "dialogue" }, "=", { "const": 4 } ]
+            },
```
Comment (GuardianDll) [inline data/json/npcs/random_encounters/camper_van_traveler.json:261]:
```suggestion
            { "math": [ "npc_randomize_dialogue_direction", "==", "4" ] },
```

File: data/json/encounters/randec_independent_travelers.json [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+    "condition": { "not": { "u_has_var": "picked_direction", "type": "dialogue", "context": "travel_direction", "value": "yes" } },
+    "effect": [
+      {
+        "//": "Most people aren't going east because of the big cities with lots of zombies, so that response isn't as likely to come up.",
+        "weighted_list_eocs": [
+          [
+            {
+              "id": "going_north",
+              "effect": [
+                { "u_add_var": "going_north", "type": "dialogue", "context": "travel_direction", "value": "yes" },
```
Comment (Procyonae) [inline data/json/encounters/randec_independent_travelers.json:34]:
```suggestion
                { "u_add_var": "going_north", "type": "dialogue", "context": "travel_direction", "value": "yes" },
```
Couldn't you change these to a single variable with values corresponding to the compass direction to make the check below simpler and more readable?

**Pull request merged**